src/LCRmeter-Keithley_4200-SCS/main.py

```diff
@@ -0,0 +1,348 @@
+        self.port_string: str = ""
+
+        # block below is needed if KXCI handling will be implemented in future
+        # where commands can be sent via GPIB or TCPIP
+        # if not RUNNING_ON_4200SCS:
+
+        # self.port_types = ["GPIB", "TCPIP"]
+        # needed to make the code working with pysweepme where port_manager is only used from __init__
+        # self.port_manager = True
+
+        # self.port_properties = {
+        #     "EOL": "\r\n",
+        #     "timeout": 10.0,
+        #     "TCPIP_EOLwrite": "\x00",
+        #     "TCPIP_EOLread": "\x00",
+        # }
+
+        # Currently, only auto range is used
+        self.range: str = ""
+        self.current_ranges = {
+            "Auto": 0,
+        }
+
+        self.operating_mode: str = ""
+        self.operating_modes = {
+            "ZTH": 0,  # "KI_CVU_TYPE_ZTH"
+            "RjX": 1,  # "KI_CVU_TYPE_RJX"
+            "CpGp": 2,  # "KI_CVU_TYPE_CPGP"
+            "CsRs": 3,  # "KI_CVU_TYPE_CSRS"
+            "CpD": 4,  # "KI_CVU_TYPE_CPD"
+            "CsD": 5,  # "KI_CVU_TYPE_CSD"
+            "YTH": 7,  # "KI_CVU_TYPE_YTH"
+        }
+
+        self.speed: str = ""
+        self.speeds = {
+            "Fast": 0,  # "KI_CVU_SPEED_FAST"
+            "Normal": 1,  # "KI_CVU_SPEED_NORMAL",
+            "Quiet": 2,  # "KI_CVU_SPEED_QUIET",
+        }
+
+        # Device specific properties
+        self.frequency: int = 0
+
+        # in case there is more than one CVU card possible, we need to use the option "Channel" which however
+        # does not yet exist for the LCRmeter module and needs to be added in this case
+        self.card_name = "CVU1"
+        self.card_id: int = 0  # placeholder value
+
+        self.measured_dc_bias: float = 0.0
+        self.measured_frequency: float = 0.0
+        self.resistance: float = 0.0
+        self.reactance: float = 0.0
+
+        self.bias_mode: str = ""
+        self.value_level_rms = None
+        self.value_bias = None
+        self.trigger_type = None
+
+        self.trigger_delay: float = 0.0
+        self.integration = None
+        self.ALC = None
+        self.stepmode = None
+        self.sweepmode = None
+        self.shortname = "Keithley 4200-SCS"
+
+    def find_ports(self) -> list:
+        """Finds the available ports for the Keithley 4200-SCS LCRmeter."""
+        ports = ["LPTlib via xxx.xxx.xxx.xxx"]
+
+        if RUNNING_ON_4200SCS:
+            ports.insert(0, "LPTlib control - no port required")
+
+        return ports
+
+    def set_GUIparameter(self) -> dict:  # noqa: N802
+        """Set initial GUI parameter in SweepMe!."""
+        return {
+            "SweepMode": ["None", "Frequency in Hz", "Voltage bias in V"],
+            "StepMode": ["None", "Frequency in Hz", "Voltage bias in V"],
+            "ValueRMS": 0.02,
+            "ValueBias": 0.0,
+            "Frequency": 1000.0,
+            "Integration": list(self.speeds),
+            "ALC": ["Off"],  # Currently unused
```

Comment:
If ALC is not used at all it could be also commented out so that the the user interface for this field gets greyed out.